pkg/repository/OCIWatcher.go

```diff
@@ -0,0 +1,182 @@
+/*
+Copyright 2023 The Kubebb Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package repository
+
+import (
+	"fmt"
+	v1 "k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/api/errors"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/util/wait"
+	"k8s.io/cli-runtime/pkg/genericclioptions"
+	ctrl "sigs.k8s.io/controller-runtime"
+	"sigs.k8s.io/controller-runtime/pkg/controller/controllerutil"
+	"strings"
+
+	"github.com/kubebb/core/api/v1alpha1"
+	"github.com/kubebb/core/pkg/helm"
+)
+
+var _ IWatcher = (*OCIWatcher)(nil)
+
+func NewOCIWatcher(watcher Watcher) IWatcher {
+	return &OCIWatcher{
+		Watcher: watcher,
+	}
+}
+
+type OCIWatcher struct {
+	Watcher
+}
+
+func (c *OCIWatcher) Start() error {
+	username, password, err := Start(c.ctx, c.instance, c.duration, c.repoName, c.c, c.logger)
+	if err != nil {
+		return err
+	}
+	c.username = username
+	c.password = password
+
+	if err := helm.RepoAddOCI(c.ctx, c.logger, c.repoName, c.instance.Spec.URL, c.username, c.password, c.duration/2); err != nil {
+		c.logger.Error(err, "Failed to add repository")
+		return err
+	}
+	if err := helm.RepoUpdateOCI(c.ctx, c.logger, c.repoName, c.duration/2); err != nil {
+		c.logger.Error(err, "Failed to update repository")
+		return err
+	}
+
+	go wait.Until(c.Poll, c.duration, c.ctx.Done())
+	return nil
+}
+
+// Poll the components
+func (c *OCIWatcher) Poll() {
+	c.logger.Info("OCIWatcher poll")
+	now := metav1.Now()
+	readyCond := getReadyCond(now)
+	syncCond := getSyncCond(now)
+
+	if err := helm.RepoUpdateOCI(c.ctx, c.logger, c.repoName, c.duration/2); err != nil {
+		c.logger.Error(err, "Failed to update repository")
+		return
+	}
+	combinedName := strings.Split(c.instance.Spec.URL, "/")
```

Comment:
get combinedName and entryName can become a util function?

Reply:
Good idea! Function getOCIEntryName() is now in utils/path.go. The changes will be uploaded in the next commit!